### DISCUSSION TITLE
How to define type for the data?

### DISCUSSION THREAD

**Discussion by alamenai** [recharts/recharts#3663] [Q&A] [answered]
Hello everyone,

First of all, thank you so much for such a kind of this library.

Because this is the first time of usage, I just get confused about how to create customizable data type, I meant that data type ( array ) can be changed based on the objects.

#### index.tsx

```typescript

import { IProps } from "./type";

import { LineChart, Line, CartesianGrid, XAxis, YAxis } from "recharts";

const Chart = <T extends { [key: string]: Array<T> }>({ data }: IProps<T>) => {
  return (
    <LineChart
      width={600}
      height={300}
      data={data}
    >
      <Line
        type="monotone"
        dataKey="uv"
        stroke="#8884d8"
      />
      <CartesianGrid stroke="#ccc" />
      <XAxis dataKey="name" />
      <YAxis />
    </LineChart>
  );
};

export default Chart;

```

#### type.ts

```typescript

export interface IProps<T> {
  data: Array<T>;
}

```

Is it right, what I made?

Thank you,

**Comment by ckifer** [comment, marked as answer]
That looks fine sure - the datatype in recharts does not currently accept a generic and is just `Array<any>` (product of this previously being a JS library). However you want to type your data outside of recharts should be fine
